v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -34,14 +34,10 @@ def from_dataset_configs(
         rdc: ReferenceDatasetCollection,
         dataset_type: DatasetType,
         reference_genome: ReferenceGenome,
-        single_dataset: str | None = None,
+        datasets: list[str] | None = None,
```

Comment:
sorry to push harder on this... I think we want either `datasets` or `rdc` but not both here.  `rdc`  being passed is basically just a proxy for the list of `datasets` to build.

honestly what we really want is for the `CachedReferenceDatasetQuery` object to be passed in here instead of an `rdc`.  So maybe something like:

```
    @classmethod
    def from_dataset_configs(
        cls,
         dc: DatasetCollection,
         dataset_type: DatasetType,
         reference_genome: ReferenceGenome,

...
    for dataset in dc.datasets(dataset_type):
```

with both `CachedReferenceDatasetQuery` and `ReferenceDatasetCollection` implementing the `DatasetCollection` interface (a "datasets" method).

Reply:
This is a little more of a broader change, so lemme know if you want to chat about it.

Reply:
And it's also annoying because the `CachedReferenceDatasetQuery` has `dataset` instead of `datasets` now, but it's probably a good idea to have the behavior be shared.

Reply:
this makes sense and shouldn't be too big of a change!

Reply:
👍 that seems reasonable!